cassiopeia/data/CassiopeiaTree.py

```diff
@@ -1874,4 +1929,4 @@ def get_distances(
 
     def copy(self) -> "CassiopeiaTree":
         """Full copy of CassiopeiaTree"""
-        return copy.deepcopy(self)
\ No newline at end of file
+        return copy.deepcopy(self)
```

Comment:
This is unrelated to this PR, but I think the standard way to implement copy and deepcopy in a class is by implementing the `__copy__` and `__deepcopy__` methods. In other words here we want to define `__deepcopy__`, not `copy`. See https://docs.python.org/3/library/copy.html , specifically:

_In order for a class to define its own copy implementation, it can define special methods \_\_copy\_\_() and \_\_deepcopy\_\_(). The former is called to implement the shallow copy operation; no additional arguments are passed. The latter is called to implement the deep copy operation; it is passed one argument, the memo dictionary. If the \_\_deepcopy\_\_() implementation needs to make a deep copy of a component, it should call the deepcopy() function with the component as first argument and the memo dictionary as second argument._

Reply:
This is a good point. I'm going to make an issue for this.